testsuite/python/mpiio.py

```diff
@@ -0,0 +1,150 @@
+#
+# Copyright (C) 2013-2019 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""
+Testmodule for MPI-IO.
+"""
+
+from __future__ import print_function
+import espressomd
+import espressomd.io
+from espressomd.interactions import AngleHarmonic
+import numpy
+import unittest as ut
+import random
+import os
+from argparse import Namespace
+#from tests_common import abspath
+
+# Number of particles
+npart = 1023
+# Number of different bond types
+nbonds = 100
+
+filename = "testdata.mpiio"
+exts = ["head", "pref", "id", "type", "pos", "vel", "boff", "bond"]
+filenames = [filename + "." + ext for ext in exts]
+
+
+def clean_files():
+    for f in filenames:
+        if os.path.isfile(f):
+            os.remove(f)
+
+
+def randint_different_from(a, b, n):
+    """Returns a random integer in [a, b) that is not n."""
+    r = n
+    while r == n:
+        r = random.randint(a, b - 1)
+    return r
+
+
+def random_particles():
+    """Returns a list of random particle descriptions."""
+    parts = []
+    for i in range(npart):
+        p = Namespace()
+        p.id = i
+        p.type = random.randint(0, 100)
+        p.pos = numpy.random.rand(3)
+        p.v = numpy.random.rand(3)
+        p.bonds = []
+        # Up to 20 bonds; otherwise this test will take ages
+        for _ in range(random.randint(0, 20)):
+            btype = random.randint(0, nbonds - 1)
+            # Don't create loops, i.e. exclude "i" itself
+            p1 = randint_different_from(0, npart, i)
+            p2 = randint_different_from(0, npart, i)
+            p.bonds.append((btype, p1, p2))
+        parts.append(p)
+    return parts
+
+
+class MPIIOTest(ut.TestCase):
+
+    """
+    Test class for the MPI-IO core functionality.
+    Generates random particles, dumps them, reads them in,
+    again and then checks the input against the initially created random
+    particles.
+    """
+    s = espressomd.system.System(box_l=[1, 1, 1])
+    # Just a bunch of random interaction such that add_bond does not throw
+    for i in range(nbonds):
+        s.bonded_inter[i] = AngleHarmonic(bend=i, phi0=i)
+    test_particles = random_particles()
+
+    def setup_sample_system(self):
```

Comment:
could you please use the python unittest class' special functions like ```setUp``` etc. for preparing and cleanup?